opm/autodiff/BlackoilWellModel_impl.hpp

```diff
@@ -1189,9 +1196,9 @@ namespace Opm {
                             // observed phase rates translated to
                             // reservoir conditions.  Recall sign
                             // convention: Negative for producers.
-                            const double target =
-                                - std::inner_product(distr.begin(), distr.end(),
-                                                     hrates.begin(), 0.0);
+                            std::vector<double> hrates_resv(np);
+                            rateConverter_->calcReservoirVoidageRates(fipreg, pvtreg, hrates, hrates_resv);
+                            const double target = -std::accumulate(hrates_resv.begin(), hrates_resv.end(), 0.0);
```

Comment:
I do not have problem with this, just some information from our experience. Without the change in the function `calcReservoirVoidageRates()`, the two functions will get the same result for the RESV.  But the good thing for `calcReservoirVoidageRates()` is that we also get the voidage rates for separate phases.

Reply:
One thing here is that if we change the way (slightly) to update the control target for `RESV`, should we also change the way to calculate the the voidage rates during the simulation for consistence?  That one can be apparently more work though.

Reply:
If the change related to rs and rv is agreed, maybe one solution is to apply the same change to the function `calcCoeff`, then we can use the old way for the `RESV` related.

Reply:
I think the code related to RESV needs to be revised altogether. Currently there are several places where we convert between surface and reservoir rates. Why not just keep the reservoir rates as they are computed computePerfRate(...) Effective Rs and Rv values can then be computed for the well based on these rates. I also think we should stop using distr for RESV wells and instead to the conversion directly using the rateConverter_ 
Maybe we can find time to discuss this at our meeting in Oslo